.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+#main.yml
+
+name: Main
+on: pull_request
+
+jobs:
+  checks:
+    runs-on: macOS-latest
+    steps:
+
+      - name: Project Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Compute key for CI cache
+        run: ./.github/workflows/compute-ci-cache-key.sh key.txt
+
+      - name: Setup cache
+        uses: actions/cache@v2.1.3
+        continue-on-error: true
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('key.txt') }}
+          restore-keys: ${{ runner.os }}-gradle-
+
+      - name: Setup JDK 1.8
+        uses: actions/setup-java@v1.4.3
+        with:
+          java-version: 1.8
+
+      - name: Clean project
+        run: ./gradlew clean
```

Comment:
This clean step shouldn't be necessary, as mentioned below. Each workflow run starts with no build files.

Reply:
Cleaned here https://github.com/square/leakcanary/pull/2012/commits/f4259b59614fecbd4b5f062d91b2b6c150f2ccb1

(no pun intended 😂  )